cmd/podman/machine/machine.go

```diff
@@ -64,3 +96,63 @@ func getMachines(toComplete string) ([]string, cobra.ShellCompDirective) {
 	}
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
+
+func initMachineEvents(cmd *cobra.Command, _ []string) error {
+	logrus.Debugf("Called machine %s.PersistentPreRunE(%s)", cmd.Name(), strings.Join(os.Args, " "))
+
+	sockPaths, err := resolveEventSock()
+	if err != nil {
+		return err
+	}
+
+	// No sockets found, so no need to publish events...
+	if len(sockPaths) == 0 {
+		return nil
+	}
+
+	for _, path := range sockPaths {
+		conn, err := (&net.Dialer{}).DialContext(registry.Context(), "unix", path)
+		if err != nil {
+			return err
+		}
+		logrus.Debugf("Machine event socket %q found", path)
+		sockets = append(sockets, conn)
+	}
+	return nil
+}
+
+func resolveEventSock() ([]string, error) {
+	// Used mostly for testing
+	if sock, found := os.LookupEnv("PODMAN_EVENTS_SOCK"); found {
+		return []string{sock}, nil
+	}
+
+	xdg, err := util.GetRuntimeDir()
+	if err != nil {
+		logrus.Warnf("Failed to get runtime dir, machine events will not be published: %s", err)
+		return nil, nil
+	}
+
+	sockPaths := make([]string, 0)
+	re := regexp.MustCompile(filepath.Join(xdg, "podman", "machine_events.*.sock"))
+	fn := func(path string, info os.DirEntry, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case info.IsDir():
+			return filepath.SkipDir
+		case info.Type() != os.ModeSocket:
+			return nil
+		case !re.MatchString(path):
```

Comment:
Could this be simplified to use info.Name() and compare to "machine_events.*.sock"